src/overlays/actors/ovl_En_Go/z_en_go.c

```diff
@@ -1091,126 +1490,138 @@ void func_80A137C0(EnGo* this, PlayState* play, f32 arg2, f32 arg3) {
-                this->unk_390 &= ~0x4000;
-                this->unk_390 &= ~0x80;
-                this->unk_390 |= 0x200;
-                this->unk_3C4++;
-                this->unk_3C2 = 0;
-                this->actor.shape.yOffset = 14.0f;
+                this->actionFlags &= ~ENGO_FLAG_ROLLING_UP;
+                this->actionFlags &= ~ENGO_FLAG_STANDING;
+                this->actionFlags |= ENGO_FLAG_ROLLED_UP;
+                this->gatekeeperAnimState++;
+                this->cutsceneDelayTimer = 0;
+                this->actor.shape.yOffset = ENGO_ROLLEDUP_Y_OFFSET;
             }
             break;
 
         case 3:
-            this->unk_3C2++;
-            if (this->unk_3C2 >= 10) {
+            this->cutsceneDelayTimer++;
+            if (this->cutsceneDelayTimer >= 10) {
                 Actor_PlaySfx(&this->actor, NA_SE_EN_GOLON_JUMP);
-                this->actor.velocity.y = 10.0f;
+                this->actor.velocity.y = ENGO_POUND_INITIAL_RISE_VEL;
                 this->actor.gravity = -1.0f;
-                this->unk_3C4++;
+                this->gatekeeperAnimState++;
             }
             break;
 
         case 4:
-            if (func_80A13564(this, 10.0f, 0.004f, 6)) {
+            if (EnGo_HandleGatekeeperPoundCutscene(this, ENGO_POUND_INITIAL_RISE_VEL, (0.4f) * ENGO_NORMAL_SCALE, 6)) {
                 Actor_PlaySfx(&this->actor, NA_SE_EN_GOLON_LAND_BIG);
-                func_80A13728(this, play);
-                this->unk_3C4++;
-                this->unk_3C2 = 0;
-                SET_WEEKEVENTREG(WEEKEVENTREG_88_40);
+                EnGo_CreateGatekeeperPoundEffects(this, play);
+                this->gatekeeperAnimState++;
+                this->cutsceneDelayTimer = 0;
+                SET_WEEKEVENTREG(WEEKEVENTREG_88_40); // Gatekeeper has Opened Shrine
             }
             break;
 
         case 5:
-            this->unk_3C2++;
-            if (this->unk_3C2 >= 10) {
-                func_80A12C48(this, play, 5);
+            this->cutsceneDelayTimer++;
+            if (this->cutsceneDelayTimer >= 10) {
+                EnGo_ChangeAnim(this, play, ENGO_ANIM_SHIVER);
                 this->actor.shape.rot.x = 0;
-                this->unk_390 &= ~0x200;
-                this->unk_390 |= 0x80;
-                this->unk_3C4++;
+                this->actionFlags &= ~ENGO_FLAG_ROLLED_UP;
+                this->actionFlags |= ENGO_FLAG_STANDING;
+                this->gatekeeperAnimState++;
             }
             break;
 
         case 6:
-            this->unk_3C0++;
-            if (this->unk_3C0 >= 65) {
+            this->cutsceneState++;
+            if (this->cutsceneState >= 65) {
                 switch (player->transformation) {
                     case PLAYER_FORM_HUMAN:
-                        SET_WEEKEVENTREG(WEEKEVENTREG_88_80);
+                        SET_WEEKEVENTREG(WEEKEVENTREG_88_80); // Gatekeeper has opened shrine for Human form.
```

Comment:
It would be nice to name this `WEEKEVENTREG`, but if you don't want to, it would be better to move these comments to the definition of this in `z64save.h`